### PR TITLE
change const DEFAULT_PVC_SIZE from 20 to 20Gi as in frontend notebook const ts

### PR DESCRIPTION
**Pull request by shalberd**
closes: #1333

## Description

The PVC Size Default in odh dashboard code is set to 20 in both backend and frontend code.

It should be taken from a default entry in odh dashboard config and not be set to a value explicitely in all sorts of places frontend and backend. There is always an auto-generated odh dashboard config, and pvc size is non-optional.
https://github.com/search?q=repo%3Aopendatahub-io%2Fodh-dashboard%20DEFAULT_PVC_SIZE&type=code

In Jupyter tile, the error with number 20 leads to a PVC being pending forever, with event message saying 

**"20 bytes is not a valid size"**

Default should be set to 20Gi.

This is only happening if no other value is specified in OdhDashboardConfig odh-dashboard-config at notebook controller level. 

## How Has This Been Tested?
set this manually in OdhDashboardConfig to 20Gi and then PVC was created correctly in both Jupyter tile and Data science projects workbenches.

## Test Impact
install ODH
do not set in ODH Dashboard GUI or OdhDashboardConfig any default pvc size
see error appearing and, after merging this PR, not appearing anymore

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by shalberd** [comment]
@DaoDaoNoCode can you look at this, please? I remember when I had not set any value in OdhDashboardConfig, I got that error "20 is not a number". 

On the other hand, I see in other places that Gi is added during assemblePvC to number

https://github.com/opendatahub-io/odh-dashboard/blob/3216f87b0762bead806c3ea377c73a0338f524bf/frontend/src/api/k8s/pvcs.ts#L36

so I really don't know anymore what is the correct way to address this inconsistency, need input from someone with insights into code structure. It might be the error "20 is not a number" only occurred on the jupyter tile.

File: frontend/src/pages/clusterSettings/const.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { ClusterSettingsType } from '~/types';
 
-export const DEFAULT_PVC_SIZE = 20;
+export const DEFAULT_PVC_SIZE = '20Gi';
```
Comment (DaoDaoNoCode) [inline frontend/src/pages/clusterSettings/const.ts]: Same here, this change is not needed.

**Review by DaoDaoNoCode — CHANGES_REQUESTED** [review]

**Comment by shalberd** [comment]
@DaoDaoNoCode ok, reworked, kept all as numbers, only modifed backend / jupyter tile pvc assembly logic as in frontend, with Gi

File: backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ const jupyterhubCfg = 'jupyterhub-cfg';
 const nbcCfg = 'notebook-controller-culler-config';
 const segmentKeyCfg = 'odh-segment-key-config';
 
-const DEFAULT_PVC_SIZE = 20;
+const DEFAULT_PVC_SIZE = '20Gi';
```
[the remark this answers is not in the document]
Comment (shalberd) [reply backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts]: yes, changed back

**Comment by shalberd** [comment]
question is if pvcSize for jupyter tile in backend, if taken from the dashboard config, is also stripped of Gib is there is such a string in the dashboardconfig.

**Comment by shalberd** [comment]
@DaoDaoNoCode ok, and keep backend notebookUtils assemblePVC as is

https://github.com/opendatahub-io/odh-dashboard/blob/3216f87b0762bead806c3ea377c73a0338f524bf/backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts#L147

**Comment by shalberd** [comment]
Yeah makes sense cause the old logic assumes pvcSize to be a string

` const pvcSize = getDashboardConfig().spec?.notebookController?.pvcSize ?? DEFAULT_PVC_SIZE;`

**Comment by shalberd** [comment]
good, changed

**Comment by DaoDaoNoCode** [comment]
@shalberd If it's done, can you remove the `WIP` in the title?

**Comment by shalberd** [comment]
@DaoDaoNoCode ah, yes, sorry, forgot :-) Thank you

**Review by lucferbux — APPROVED** [review]
/lgtm

**Pull request merged**